### PR TITLE
Feature: Adding method to get order ingredients by order id

### PR DESCRIPTION
**Pull request by eapg**
* Method to get order ingredients by order id

This method will get the ingredients of an order by its id. with the order id it will get all the order_detail(product) associate with that id calling the method inside order_detail_repository **get_by_order_id** , after get the order details we can get the product_ingredients(ingredient of a product with its quantity needed) looping the order_details. 

It is important to highlight that the order_detail_repository and product_ingredient_repository must be sent in the constructor of the order_repository  to be able to do all the operations.

### CODE REVIEW

File: src/lib/repositories/impl/order_repository_impl.py [hunk 1]
```diff
@@ -30,5 +34,23 @@ def update_by_id(self, order_id, order):
...
+            lambda order: order.status == OrderStatus.NEW_ORDER, orders
+        )
         return list(orders_to_process)
+
+    def get_order_ingredients_by_order_id(self, order_id):
+
+        order_ingredients = []
+        order_details = self.order_detail_repository.get_by_order_id(order_id)
+
+        for order_detail in order_details:
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/order_repository_impl.py:47]: Try to map order details to an array of product ids.

Create a function in product ingredient repository that accepts a list of product ids and return an array of product ingredients for those product ids.

You don't need to multiply elements by quantity

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/lib/repositories/impl/order_repository_impl.py (hunk 1 above)
Comment (eapg) [reply src/lib/repositories/impl/order_repository_impl.py]: done

File: src/lib/repositories/impl/product_ingredient_repository_impl.py [hunk 2]
```diff
@@ -43,3 +43,11 @@ def get_by_product_id(self, product_id):
             product_ingredients,
         )
         return list(product_ingredients_of_product)
+
+    def get_product_ingredients_by_products_id(self, products_id):
+        product_ingredients = []
+
+        for product_id in products_id:
+            product_ingredients.extend(self.get_by_product_id(product_id))
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/product_ingredient_repository_impl.py:51]: Get all products and then use filter

File: src/lib/repositories/impl/order_repository_impl.py [hunk 3]
```diff
@@ -30,5 +34,18 @@ def update_by_id(self, order_id, order):
...
-        orders_to_process = filter(lambda order: order.status == OrderStatus.NEW_ORDER, orders)
+        orders_to_process = filter(
+            lambda order: order.status == OrderStatus.NEW_ORDER, orders
+        )
         return list(orders_to_process)
+
+    def get_order_ingredients_by_order_id(self, order_id):
+
+        order_details = self.order_detail_repository.get_by_order_id(order_id)
+        products_id = [order_detail.product_id for order_detail in order_details]
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/order_repository_impl.py:45]: Rename this to product_ids

**Review by reynoldmorel — APPROVED** [review]

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
Changing to right state

File: src/lib/repositories/impl/product_ingredient_repository_impl.py (hunk 2 above)
Comment (eapg) [reply src/lib/repositories/impl/product_ingredient_repository_impl.py]: done.

File: src/lib/repositories/impl/order_repository_impl.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories/impl/order_repository_impl.py]: done

File: src/lib/repositories/impl/product_ingredient_repository_impl.py [hunk 4]
```diff
@@ -43,3 +43,13 @@ def get_by_product_id(self, product_id):
             product_ingredients,
         )
         return list(product_ingredients_of_product)
+
+    def get_product_ingredients_by_product_ids(self, product_ids):
+
+        product_ingredients = self.get_all()
+        product_ingredients_by_product_ids = filter(
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/product_ingredient_repository_impl.py:50]: `filtered_product_ingredients`?

File: src/lib/repositories/impl/order_repository_impl.py [hunk 5]
```diff
@@ -30,5 +34,18 @@ def update_by_id(self, order_id, order):
...
+        orders_to_process = filter(
+            lambda order: order.status == OrderStatus.NEW_ORDER, orders
+        )
         return list(orders_to_process)
+
+    def get_order_ingredients_by_order_id(self, order_id):
+
+        order_details = self.order_detail_repository.get_by_order_id(order_id)
+        product_ids = [order_detail.product_id for order_detail in order_details]
+        product_ingredients_list = (
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/order_repository_impl.py:46]: `filtered_product_ingredients`?

File: src/tests/lib/repositories/impl/order_repository_impl_test.py [hunk 6]
```diff
@@ -119,3 +129,38 @@ def test_get_orders_to_process(self):
...
+        )
+        order_detail_repository.add(order_detail_1)
+        order_repository = OrderRepositoryImpl(
+            order_detail_repository, product_ingredient_repository
+        )
+        order_ingredients = order_repository.get_order_ingredients_by_order_id(
+            order_1.id
+        )
+        product_ingredient_repository.get_product_ingredients_by_product_ids.assert_called()
+        order_detail_repository.get_by_order_id.assert_called()
```
Comment (reynoldmorel) [inline src/tests/lib/repositories/impl/order_repository_impl_test.py:162]: `assert_called_with(...)`?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
`test_get_orders_to_process(self)`

You should confirm the one that is missing is actually the one that is completed. Maybe use filter and assert that element does not exist in the array

File: src/lib/repositories/impl/product_ingredient_repository_impl.py (hunk 4 above)
Comment (eapg) [reply src/lib/repositories/impl/product_ingredient_repository_impl.py:50]: done

File: src/lib/repositories/impl/order_repository_impl.py (hunk 5 above)
Comment (eapg) [reply src/lib/repositories/impl/order_repository_impl.py:46]: done

File: src/tests/lib/repositories/impl/order_repository_impl_test.py (hunk 6 above)
Comment (eapg) [reply src/tests/lib/repositories/impl/order_repository_impl_test.py]: done

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**